packages/core/src/common/types.ts

```diff
@@ -17,6 +17,8 @@
 export { ArrayUtils } from './array-utils';
 export { Prioritizeable } from './prioritizeable';
 
+type UnknownObject = Record<string | number | symbol, unknown>;
```

Comment:
Minor: Improvements to the typing of `UnknownObject` and `isObject`. I feel like this would be a more intuitive way of understanding the return type of `isObject`.

```suggestion
type UnknownObject<T extends object> = Record<string | number | symbol, unknown> & { [K in keyof T]: unknown };
```

Reply:
It does look better, thanks for the suggestion.